internal/provider/resource_cluster.go

```diff
@@ -814,24 +814,35 @@ func (r *ClusterResource) Update(ctx context.Context, req resource.UpdateRequest
 	}
 
 	// immutable fields
-	if previous.ClusterName == nil && updated.ClusterName != nil {
+	if previous.Tier != updated.Tier {
 		resp.Diagnostics.AddError(
 			"Cannot update immutable field",
-			fmt.Sprintf("Cluster name cannot be changed, previous is not set, now is %s", *updated.ClusterName),
+			"Tier cannot be changed",
 		)
 	}
-	if previous.ClusterName != nil && updated.ClusterName == nil {
-		resp.Diagnostics.AddError(
-			"Cannot update immutable field",
-			fmt.Sprintf("Cluster name cannot be changed, previous is %s, now is not set", *previous.ClusterName),
-		)
-	}
-	if (previous.ClusterName != nil && updated.ClusterName != nil) && (*previous.ClusterName != *updated.ClusterName) {
-		resp.Diagnostics.AddError(
-			"Cannot update immutable field",
-			fmt.Sprintf("Cluster name cannot be changed, previous: %s, updated: %s", *previous.ClusterName, *updated.ClusterName),
-		)
+
+	// only check clusterName for BYOC
```

Comment:
Why do we only check BYOC clusters? Hosted clusters cannot change their names either IIRC.

Reply:
There is no `clusterName` field exposed in the tf plugin data model. So the `clusterName` is not set in the `updated` object, which is transformed from the data model object submitted by `tf plan`. The `previous` object is created from the GET tenants API data directly. So `clusterName` is set in `previous` object. 

In short, `clusterName` is not exposed to the tf plugin, the value is not set so we need to ignore it, and users cannot change it via tf plugin.